### PR TITLE
Add graders into submissions table

### LINKED ISSUE #4447: Add graders column to submission table

### PR DESCRIPTION
**Pull request by ekowidianto**
Closes #4447 
As requested by Prof.

image

image

### CODE REVIEW

**Comment by ekowidianto** [comment]
Can be reviewed after MUI update is merged.

**Comment by ekowidianto** [comment]
@zhuhanming To prioritise this PR.

File: app/models/course/assessment/submission.rb [hunk 1]
```diff
@@ -79,10 +79,6 @@ class Course::Assessment::Submission < ApplicationRecord
   has_many :question_bundle_assignments, class_name: Course::Assessment::QuestionBundleAssignment.name,
                                          inverse_of: :submission, dependent: :destroy
 
-  # @!attribute [r] graders
-  #   The graders associated with this submission.
-  has_many :graders, through: :answers, class_name: User.name
```
Comment (zhuhanming) [inline app/models/course/assessment/submission.rb:84]: Ooo why was this removed?

File: app/views/course/assessment/submission/submissions/index.json.jbuilder [hunk 2]
```diff
@@ -38,6 +39,13 @@ json.submissions @course_users do |course_user|
     json.dateSubmitted submission.submitted_at&.iso8601
     json.dateGraded submission.graded_at&.iso8601
     json.logCount submission.log_count
+    json.graders do
+      json.array! submission.grader_ids do |grader_id|
+        cu = course_users_hash[grader_id] || [0, 'Undefined']
```
Comment (zhuhanming) [inline app/views/course/assessment/submission/submissions/index.json.jbuilder:44]: Can we have it as "Unknown" or just default to "System" instead?

**Review by zhuhanming — APPROVED** [review]
LGTM! Thanks for rebasing this off the MUI upgrade branch

File: app/models/course/assessment/submission.rb (hunk 1 above)
Comment (ekowidianto) [reply app/models/course/assessment/submission.rb:84]: That’s not used at all in the codebase and getting graders using this is very slow..

File: app/views/course/assessment/submission/submissions/index.json.jbuilder (hunk 2 above)
Comment (ekowidianto) [reply app/views/course/assessment/submission/submissions/index.json.jbuilder:44]: The purpose for this is when a course_user does not exist anymore. So i think we can use unknown instead.

File: app/models/course/assessment/submission.rb (hunk 1 above)
Comment (zhuhanming) [reply app/models/course/assessment/submission.rb:84]: Roger, thanks for clarifying

**Comment by ekowidianto** [comment]
@cysjonathan ready for deployment
